firmware/library/L1_Drivers/pin_configure.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <assert.h>
+#include <cstdio>
+
+#include "L0_LowLevel/LPC40xx.h"
+#include "L2_Utilities/macros.hpp"
+
+class PinConfigureInterface
+{
+   public:
+    enum PinMode : uint8_t
+    {
+        kInactive = 0,
+        kPullDown,
+        kPullUp,
+        kRepeater
+    };
+    virtual void SetPinFunction(uint8_t function)         = 0;
+    virtual void SetPinMode(PinMode mode)                 = 0;
+    virtual void EnableHysteresis(bool enable_hysteresis) = 0;
+    virtual void SetAsActiveLow(bool set_as_active_low)   = 0;
+    virtual void EnableFastMode(bool enable_fast_mode)    = 0;
+    virtual void SetAsOpenDrain(bool set_as_open_drain)   = 0;
+};
+
+class PinConfigure : public PinConfigureInterface
+{
+   public:
+    enum PinBitMap : uint8_t
+    {
+        kFunction    = 0,
+        kMode        = 3,
+        kHysteresis  = 5,
+        kInputInvert = 6,
+        kSlew        = 9,
+        kOpenDrain   = 10
+    };
+
+    SJ2_PACKED(struct) PinMap_t
+    {
+        using pin_config_register_t = uint32_t;
+        pin_config_register_t _register[6][32];
+    };
+
+    static PinMap_t * pin_map;
+
+    constexpr PinConfigure(const uint8_t _port, const uint8_t _pin)
+        : port(_port), pin(_pin)
+    {
+        assert(_port <= 5 && "Port must be between 0 and 5");
+        assert(_pin <= 31 && "Pin must be between 0 and 31");
+    }
+    void SetPinFunction(uint8_t function) final override
+    {
+        pin_map->_register[port][pin] =
+            BitPlace(pin_map->_register[port][pin], PinBitMap::kFunction,
+                     function & 0b111, 3);
+    }
+    void SetPinMode(PinConfigureInterface::PinMode mode) final override
+    {
+        pin_map->_register[port][pin] = BitPlace(
+            pin_map->_register[port][pin], PinBitMap::kMode, mode & 0b11, 2);
+    }
+    void EnableHysteresis(bool enable_hysteresis) final override
```

Comment:
This and the functions below should have the bool be a default parameter equal to *true*, so users do not have to enter a bool in order to do the action. This should be a valid line of code:

`pin0_1.EnableHysteresis()` or `pin0_1.SetAsOpenDrain()` rather than `pin0_1.EnableHysteresis(true)`